src/query/storage/m3/consolidators/fetch_result_map.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package consolidators
+
+import (
+	"fmt"
+
+	"github.com/m3db/m3/src/dbnode/encoding"
+	xerrors "github.com/m3db/m3/src/x/errors"
+)
+
+type fetchResultMapWrapper struct {
+	resultMap *fetchResultMap
+
+	multiErr xerrors.MultiError
+}
+
+func (w *fetchResultMapWrapper) err() error {
+	return w.multiErr.FinalError()
+}
+
+func (w *fetchResultMapWrapper) len() int {
+	return w.resultMap.Len()
+}
+
+func (w *fetchResultMapWrapper) list() []multiResultSeries {
+	result := make([]multiResultSeries, 0, w.len())
+	for _, results := range w.resultMap.Iter() {
+		result = append(result, results.value)
+	}
+
+	return result
+}
+
+func (w *fetchResultMapWrapper) get(
+	it encoding.SeriesIterator,
+) (multiResultSeries, bool) {
```

Comment:
Maybe it's worth making this potentially return an error too since you can run into an error during get(...) set(..)? I know there's an explicit ".err()" but maybe it's best to just always potentially return an error from get/set (then callers don't need to know they need to remember to call ".err()", etc?